docs/docs/dev_docs/contracts/resources/common_patterns/authwit.md

```diff
@@ -0,0 +1,224 @@
+
+In the snippet below, this is done as a separate contract call, but can also be done as part of a batch as mentioned in the [foundational concepts](./../../../../concepts/foundation/accounts/authwit.md#what-about-public).
+
+#include_code authwit_public_transfer_example /yarn-project/end-to-end/src/e2e_token_contract.test.ts typescript
+
+#### Updating approval state in Noir
+
+We have cases where we need a non-wallet contract to approve an action to be executed by another contract. One of the cases could be when making more complex defi where funds are passed along. When doing so, we need the intermediate contracts to support approving of actions on their behalf.
+
+To support this, we must implement the `is_valid_public` function as seen in the snippet below.
+
+#include_code authwit_uniswap_get /yarn-project/noir-contracts/src/contracts/uniswap_contract/src/main.nr rust
+
+It also needs a way to update those storage values. Since we want the updates to be trustless, we can compute the action based on the function inputs, and then have the contract compute the key at which it must add a `true` to approve the action.
+
+An example of this would be our Uniswap example which performs a cross chain swap on L1. In here, we both do private and public auth witnesses. Where the public is set by the uniswap L2 contract itself. In the below snippet, you can see that we compute the action hash, and then update an `approved_action` mapping with the hash as key and `true` as value. When we then call the `token_bridge` to execute afterwards, it reads this value, burns the tokens, and consumes the authentication.
+
+#include_code authwit_uniswap_set /yarn-project/noir-contracts/src/contracts/uniswap_contract/src/main.nr rust
+
+Outlining more of the `swap` flow: this simplified diagram shows how it will look for contracts that are not wallets but also need to support authentication witnesses.
+
+```mermaid
+sequenceDiagram
+    actor A as Alice
+    participant AC as Alice Account
+    participant CC as Crosschain Swap
+    participant TB as Token Bridge
+    participant T as Token
+
+    A->>AC: Swap 1000 token A to B on Uniswap L1
+    activate AC;
+    AC->>CC: Swap 1000 token A to B
+    activate CC;
+    CC->>T: unshield 1000 tokens from Alice Account to CCS
+    activate T;
+    T->>AC: Have you approved this??
+    AC-->>A: Please give me an AuthWit
+    A-->>AC: Here is AuthWit
+    AC-->>AC: Validate AuthWit
+    AC->>T: Yes
+    deactivate T;
+    CC-->>CC: Setting flag to true
+    CC->>TB: Exit 1000 tokens to CCS
+    activate TB;
+    TB->>T: Burn 1000 tokens from CCS
+    activate T;
+    T->>CC: Have you approved this?
+    CC->>T: Yes
+    T-->>T: Burn
+    deactivate T;
+    TB-->>TB: Emit L2->L1 message
+    deactivate TB;
+    CC-->>CC: Emit L2->L1 message
+    deactivate CC;
+    deactivate AC;
+```
+
+:::info **TODO**
```

Comment:
do we have the blogs now?

Reply:
Its the ones @rahul-kothari have been working on, so partially.